app/guide/components/subsection.test.js

```diff
@@ -0,0 +1,198 @@
+      'Footer description sentence 3.',
+    ],
+    button: {
+      text: 'Download Buyer’s Guide PDF',
+    },
+  },
+};
+
+const macroWrapper = `{% from 'guide/components/subsection.njk' import subsection %}
+{{ subsection(subSection) }}`;
+
+describe('subsection', () => {
+  it('should render a title if provided', (done) => {
+    const app = testHarness().createTemplateDummyApp(macroWrapper, context);
+    request(app)
+      .get('/')
+      .then((res) => {
+        const $ = cheerio.load(res.text);
+        expect($('[data-test-id="subsection-title"]').text().trim()).toEqual(context.subSection.title);
+        done();
+      });
+  });
+
+  it('should not render a title if not provided', (done) => {
+    const newContext = { ...context };
+    delete newContext.subSection.title;
+    const app = testHarness().createTemplateDummyApp(macroWrapper, newContext);
+    request(app)
+      .get('/')
+      .then((res) => {
+        const $ = cheerio.load(res.text);
+        expect($('[data-test-id="subsection-title"]').length).toEqual(0);
+        done();
+      });
+  });
+
+  it('should render a description if provided', (done) => {
+    const app = testHarness().createTemplateDummyApp(macroWrapper, context);
+    request(app)
+      .get('/')
+      .then((res) => {
+        const $ = cheerio.load(res.text);
+        const description = $('[data-test-id="subsection-description"]');
+        context.subSection.description.forEach((descriptionText, i) => {
+          expect(description.find(`div:nth-child(${i + 1})`).text().trim()).toEqual(descriptionText);
+        });
+        done();
+      });
+  });
+
+  it('should render a link within description if provided', (done) => {
+    const testCaseContext = {
+      subSection: {
+        title: 'Additional Services',
+        description: [
+          {
+            startText: 'You can learn more about the Capability Model',
+            linkText: '[link]',
+            href: 'https://www.nhs.uk/',
+            endText: '',
+          },
+        ],
+      },
+    };
+    const app = testHarness().createTemplateDummyApp(macroWrapper, testCaseContext);
+    request(app)
+      .get('/')
+      .then((res) => {
+        const $ = cheerio.load(res.text);
+        const description = $('[data-test-id="subsection-description"]');
+        const startText = description.find('span:nth-child(1)');
+        const endText = description.find('span:nth-child(3)');
+        const link = description.find('a');
+
+        expect(endText.text().trim())
+          .toEqual(testCaseContext.subSection.description[0].endText.trim());
+        expect(startText.text().trim())
+          .toEqual(testCaseContext.subSection.description[0].startText.trim());
+        expect(link.text().trim())
+          .toEqual(testCaseContext.subSection.description[0].linkText.trim());
+        expect(link.attr('href'))
+          .toEqual(testCaseContext.subSection.description[0].href.trim());
```

Comment:
I would change this to one expect that checks it all. It's hard to work out within this test what to expect.
so something along the lines of
```
expect(description.html()).toEqual(`<div>some start text <a href='some-link'>the link</a> and the end text</div>`)
```